### PR TITLE
Combat logic

### PR DESCRIPTION
**Pull request by Suedie**
Implemented damage logic
View properties not implemented

### CODE REVIEW

File: core/src/model/Entity.java [hunk 1]
```diff
@@ -75,105 +58,52 @@ public float getWidth() {
...
-    public World<IEntity> getWorld() {
-        return world;
+    public Item<IEntity> getBoundingbox() {
+        return boundingbox;
     }
 
     /**
      * Adds a reference to the world that the player is in and
-     * registers itself as a collisionbox
+     * registers itself as a collisionbox within in. Useful for
```
Comment (pkasinski) [inline core/src/model/Entity.java]: Typo.

File: core/src/model/IEnvironmentCache.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+public interface IEnvironmentCache {
+
+    /**
+     * Gets the JBump "world" which is used for moving entities and registering collisions
+     * @return the JBump world
+     */
+    World<IEntity> getWorld();
+
+    /**
+     * Used to make sure the map used by the view is the same as the model
```
Comment (pkasinski) [inline core/src/model/IEnvironmentCache.java]: I'm not sure if the view and the model should be mentioned in the documentation for this method. The documentation should say what the method does.

File: core/src/model/Model.java [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
 import model.rewards.RewardSystem;
-
 import java.util.ArrayList;
 import java.util.List;
 
 public class Model implements MovementListener {
-    private IMapLoader mapLoader;
     private IPlayerCharacter player;
-    private List<Enemy> enemyList = new ArrayList<>();
+    private List<Monster> monsterList = new ArrayList<>();
```
Comment (pkasinski) [inline core/src/model/Model.java]: Could be renamed to monsters.

File: core/src/model/Model.java [hunk 4]
```diff
@@ -28,12 +27,25 @@ public void update() {
...
         player = rewardSystem.applyReward(player, Reward.SPEED_DEVIL);
         }
+        checkEnemyHealth();
+    }
+
+    /**
+     * Goes through the list of enemies checks if they need to be removed
+     * O(n)
+     */
+    public void checkEnemyHealth() {
```
Comment (pkasinski) [inline core/src/model/Model.java]: I think this method could possibly be given a better name.

File: core/src/model/Model.java [hunk 5]
```diff
@@ -86,10 +97,20 @@ public World<IEntity> getWorld() {
...
         rewardSystem.initialize(world);
         this.mapLoader = mapLoader;
         player = new PlayerCharacter(mapLoader.getMapUnitWidth() / 2, mapLoader.getMapUnitHeight() / 2, mapLoader.getWorld());
         player.gainExperience(100);
     }
+
+    /**
+     * Removes an Enemy from the game and removes
+     * its collisionbox from the world
+     * @param monster Enemy to be removed
```
Comment (pkasinski) [inline core/src/model/Model.java:146]: Enemy is written instead of monster.

**Review by pkasinski — APPROVED** [review]
LGTM. I left a few comments.

File: core/src/model/Entity.java [hunk 6]
```diff
@@ -14,45 +14,28 @@ public abstract class Entity implements IEntity{
...
     }
-    public void setCollisionType (CollisionFilter collisionType) {
-        this.collisionType = collisionType;
+
+    public void setCollisionResponse(CollisionFilter collisionResponse) {
+        this.collisionResponse = collisionResponse;
     }
 
-     public Entity(float x, float y, float height, float width, float speed, float maxHealth , float damage, World<IEntity> world) {
+     public Entity(float x, float y, float height, float width, float damage, World<IEntity> world) {
```
Comment (rojtekin) [inline core/src/model/Entity.java:30]: maybe damage should be in only livingentity?

File: core/src/model/ModelFactory.java [hunk 7]
```diff
@@ -1,15 +1,15 @@
 package model;
 
-import model.enemies.*;
+import model.monsters.*;
```
Comment (rojtekin) [inline core/src/model/ModelFactory.java]: Every hard coded monster should be deleted in order to merge with the rest of the code

File: core/src/model/PlacedMapEntity.java [hunk 8]
```diff
@@ -6,7 +6,11 @@
...
  */
 public class PlacedMapEntity extends Entity {
-    public PlacedMapEntity(float x, float y, float height, float width, World<IEntity> world) {
-        super(x, y, height, width, 0, 1000000,0, world);
+    public PlacedMapEntity(float x, float y, float height, float width, float damage, World<IEntity> world) {
+        super(x, y, height, width, damage, world);
+    }
+
+    @Override
+    public void beAttacked(float damage, Faction faction) {
```
Comment (rojtekin) [inline core/src/model/PlacedMapEntity.java:14]: maybe the program works without beAttacked()?

**Review by rojtekin — APPROVED** [review]
LGTM

**Pull request merged**